logos-codegen/src/mir.rs

```diff
@@ -192,7 +194,7 @@ mod tests {
     fn priorities() {
         let regexes = [
             ("[a-z]+", 1),
-            ("a|b", 2),
+            ("a|b", 1),
```

Comment:
Note, the HIR representation here changes from `Alternate(['a', 'b'])` to `Class('a'..='b')`, which is probably the result of an upstream optimisation - if the test wants it to be an alternate, then it's probably better to do `a|c`.

Reply:
As discussed in #320, I don't think we just adapt our code to possible optimization. Otherwise, this can be a never ending game where each optimizer in `regex-syntax` would lead to a breaking change in `logos` :'-)

Reply:
Hum i don’t think optimisations need to be lossy. To me, an optimisation results in the same behavior, which is what the priority should compute

Reply:
I clarified what I meant as lossy as “non-invertible”. I mean that optimisations in regex-syntax may optimise in a way that we don’t have enough information to get back to the form we expect.

However we should _always_ be able to get to the same canonical DFA.

Reply:
https://en.m.wikipedia.org/wiki/DFA_minimization